### ISSUE TITLE
java.lang.NullPointerException

### ISSUE THREAD

**Issue by tinfoilhat** [tinfoilhat/tinfoil-sms#23]
``` bash
java.lang.NullPointerException
at com.tinfoil.sms.crypto.Encryption.initECEngine(Encryption.java:229)
at com.tinfoil.sms.crypto.Encryption.decrypt(Encryption.java:148)
at com.tinfoil.sms.utility.MessageReceiver.onReceive(MessageReceiver.java:169)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2424)
at android.app.ActivityThread.access$1500(ActivityThread.java:141)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:5103)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
at dalvik.system.NativeStart.main(Native Method)
java.lang.NullPointerException
at com.tinfoil.sms.crypto.Encryption.initECEngine(Encryption.java:229)
at com.tinfoil.sms.crypto.Encryption.decrypt(Encryption.java:148)
at com.tinfoil.sms.utility.MessageReceiver.onReceive(MessageReceiver.java:169)
at android.app.ActivityThread.handleReceiver(ActivityThread.java:2424)
at android.app.ActivityThread.access$1500(ActivityThread.java:141)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1332)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:137)
at android.app.ActivityThread.main(ActivityThread.java:5103)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:525)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:737)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
at dalvik.system.NativeStart.main(Native Method)

```

**Comment by gnu-user** [comment]
Was this the same NullPointerException we had tried to fix earlier where the conclusion was that the _SMSUtility.user.getPrivateKey()_ in the line:

``` java
ECPrivateKeyParameters priKey = ECGKeyUtil.decodeBase64PriKey(param, SMSUtility.user.getPrivateKey());
```

Was the cause of the error, or is this a another instance of the same type of problem that we had solved earlier? I've looked through the code thoroughly and I can't see anything else that would cause a NullPointerException.

**Comment by dataBaseError** [comment]
Are you sure the error is coming from the right version of tinfoil SMS?

**Comment by gnu-user** [comment]
It has been occurring recently with version 1.2.3, I'm not sure if this was one of the bugs that was fixed in 1.2.4, if it was then we can mark it as resolved. You can see the details of it on bugsense here:

[NullPointerException Error](https://www.bugsense.com/dashboard/project/169095e2/errors/204227304)

**Comment by dataBaseError** [comment]
The issue that shows up on Github told me that it was only for 1.2.2,1.2.3.
Also for the issue about the bad base 64, the method is not yet deprecated
since the new method doesn't exist. Also I can't find any info about SMS
messages getting un-organized, the new method is to handle more types of
networks. Finally, the only thing I think it might be but it's a bit of a
stretch, is two different number's messages getting combined into one. But
really I would need to be able to re create.

**Comment by gnu-user** [comment]
This issue has been fixed in 1d20c0cd56eac9aedf6a63266e36e61c046bdce8, the issue was that in some scenarios _SMSUtility.user.getPrivateKey()_ was returning null.
